File: DoConnect_Final/src/main/java/com/doconnect/service/interfaces/UserService.java
package com.doconnect.service.interfaces;

import java.util.List;

import com.doconnect.dto.answer.AnswerResponseDTO;
import com.doconnect.dto.question.QuestionResponseDTO;
import com.doconnect.dto.user.ChangePasswordDTO;
import com.doconnect.dto.user.UpdateProfileDTO;
import com.doconnect.dto.user.UserProfileDTO;
import com.doconnect.enums.Role;

public interface UserService { 

    UserProfileDTO getProfile(String email); 

    UserProfileDTO getUserById(Long id); 

    UserProfileDTO updateProfile( 

            String email, 

            UpdateProfileDTO dto); 

    void changePassword( 

            String email, 

            ChangePasswordDTO dto); 

    void deleteAccount(String email); 

    List<QuestionResponseDTO> getUserQuestions( 

            Long userId); 

    List<AnswerResponseDTO> getUserAnswers( 

            Long userId); 

    List<UserProfileDTO> getAllUsers(); 

    void updateUserRole(Long userId, Role role); 

    void deleteUser(Long userId); 

}

File: doconnect_frontend_final/src/services/userService.js
/** 

 * User Service 

 * Profile management API calls 

 */ 

import axiosInstance from '../api/axiosConfig'; 

 

const userService = { 

 

  getProfile: async () => { 

    const res = await axiosInstance 

                      .get('/users/profile'); 

    return res.data; 

  }, 

 

  updateProfile: async (data) => { 

    const res = await axiosInstance 

                .put('/users/profile', data); 

    return res.data; 

  }, 

 

  changePassword: async (data) => { 

    const res = await axiosInstance 

                .put('/users/password', data); 

    return res.data; 

  }, 

 

  getUserById: async (id) => { 

    const res = await axiosInstance 

                      .get(`/users/${id}`); 

    return res.data; 

  }, 

 

  getUserQuestions: async (id) => { 

    const res = await axiosInstance 

              .get(`/users/${id}/questions`); 

    return res.data; 

  }, 

 

  getUserAnswers: async (id) => { 

    const res = await axiosInstance 

                .get(`/users/${id}/answers`); 

    return res.data; 

  }, 

 

  deleteAccount: async () => { 

    const res = await axiosInstance 

                    .delete('/users/account'); 

    return res.data; 

  }, 

}; 

 

export default userService;